compiler/luci/pass/src/QuantizedModelVerifier.test.cpp

```diff
@@ -1107,29 +1119,28 @@ TEST(QuantizedModelVerifierTest, ArgMax)
 
 TEST(QuantizedModelVerifierTest, ArgMax_wrong_dimension_type_NEG)
 {
-  ArgMaxTestGraph<Type::S32> g;
-  g.init();
-  luci::QuantizeWithMinMaxPass pass(Type::FLOAT32, Type::U8, Granularity::LayerWise);
-  pass.run(g.g());
+  TEST_WITH_WRONG_TYPE(ArgMaxTestGraph<Type::S32>, Type::U8, Granularity::LayerWise, Type::S16);
+  TEST_WITH_WRONG_TYPE(ArgMaxTestGraph<Type::S32>, Type::U8, Granularity::ChannelWise, Type::S16);
+  TEST_WITH_WRONG_TYPE(ArgMaxTestGraph<Type::S32>, Type::S16, Granularity::ChannelWise, Type::U8);
 
-  g._dimension->dtype(Type::U8);
-
-  luci::QuantizedModelVerifier verifier(Type::U8, Granularity::LayerWise);
-  EXPECT_ANY_THROW(verifier.verify(g.g()));
+  TEST_WITH_WRONG_TYPE(ArgMaxTestGraph<Type::S64>, Type::U8, Granularity::LayerWise, Type::S16);
+  TEST_WITH_WRONG_TYPE(ArgMaxTestGraph<Type::S64>, Type::U8, Granularity::ChannelWise, Type::S16);
+  TEST_WITH_WRONG_TYPE(ArgMaxTestGraph<Type::S64>, Type::S16, Granularity::ChannelWise, Type::U8);
+  SUCCEED();
 }
 
-TEST(QuantizedModelVerifierTest, ArgMax_wrong_input_granularity_NEG)
-{
-  ArgMaxTestGraph<Type::S32> g;
-  g.init();
-
-  luci::QuantizeWithMinMaxPass pass(Type::FLOAT32, Type::U8, Granularity::LayerWise);
-  pass.run(g.g());
+// TODO : Add a negative test which sets wrong dtype on dimension node
```

Comment:
`TEST_WITH_WRONG_TYPE_TARGET` is needed for this.

I will make a separate PR for each of them.